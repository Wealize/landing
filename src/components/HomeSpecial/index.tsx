import React, { useRef } from 'react'
import Scroll from 'react-scroll'

import RoundArrowIcon from '../icons/RoundArrowIcon'
import {
  Container,
  Background,
  Button,
  Headline,
  Slogan,
  SubHeadline
} from './styles'
import WStaticIcon from '../icons/WStaticIcon'

const HomeSpecial = () => {
  const containerRef = useRef(null)
  const buttonScrollRef = useRef(null)
  const BREAKPOINT_LG_PX = 992

  const scrollToNextElement = () => {
    const navbarHeight = document.getElementById('navbar').clientHeight
    const buttonMiddleHeight = buttonScrollRef.current.clientHeight / 2

    if (!navbarHeight) return false

    if (window.innerWidth < BREAKPOINT_LG_PX) {
      Scroll.animateScroll.scrollTo(
        containerRef.current.clientHeight + navbarHeight + 2,
        { duration: 1000, delay: 0, smooth: true }
      )
    } else {
      Scroll.animateScroll.scrollTo(
        containerRef.current.clientHeight + navbarHeight + buttonMiddleHeight,
        { duration: 1000, delay: 0, smooth: true }
      )
    }
  }

  return (
    <Container ref={containerRef}>
      <Background>
        <WStaticIcon />
      </Background>
      <Slogan>
        <Headline>Unlock todayâ€™s world</Headline>
        <SubHeadline>
          Crafting digital products with real-world impact
        </SubHeadline>
        <Button
          onClick={scrollToNextElement}
          aria-label="scroll-button"
          ref={buttonScrollRef}
        >
          <RoundArrowIcon />
        </Button>
      </Slogan>
    </Container>
  )
}

export default React.memo(HomeSpecial)
