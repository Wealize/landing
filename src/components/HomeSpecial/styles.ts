import styled from 'styled-components'

import {
  PRIMARY_COLOR,
  WHITE_COLOR,
  LIGHT_GRAY_COLOR,
  ACCENT_COLOR
} from '../../theme/color'
import {
  twoThirdPx,
  doubleAndHalfPx,
  triplePx
} from '../../theme/space'
import { media } from '../../theme/media'
import { DEFAULT_RADIUS } from '../../theme/border'
import { CONTAINER_SPACING } from '../../theme/spacingGuide'

export const Container = styled.div`
  width: 100%;
  min-height: 85vh;
  background: ${LIGHT_GRAY_COLOR};
  padding: ${doubleAndHalfPx()} 0;
  position: relative;

  ${media.greaterThan('md')`
    min-height: 90vh;
  `};

  ${media.greaterThan('lg')`
    min-height: 80vh;
  `};
`

export const Background = styled.div`
  width: 100%;
  height: 70%;
  position: absolute;
  bottom: 0%;
  height: 100%;

  background-image: url('/img/w-static.svg');
  background-repeat: no-repeat;
  background-position: -25% 100%;
  background-size: 800px;

  ${media.greaterThan('md')`
    background-position: -100% 120%;
    background-size: 140%;
  `};

  ${media.greaterThan('lg')`
    background-position: -100% -20%;
    background-size: 120%;
  `};

  ${media.greaterThan('xl')`
    background-position: 230% -15%;
    background-size: 85%;
  `};

  ${media.greaterThan('xxl')`
    background-position: 180% -15%;
    background-size: 80%;
  `};
`

export const Button = styled.button`
  background-color: ${ACCENT_COLOR};
  color: ${WHITE_COLOR};
  width: 90px;
  height: 90px;
  border: none;
  border-radius: 50%;
  position: absolute;
  bottom: 82%;
  right: 10%;
  cursor: pointer;
  z-index: 2;

  ${media.greaterThan('md')`
    bottom: 85%;
    width: 115px;
    height: 115px;
  `};

  ${media.greaterThan('lg')`
    bottom: -10%;
    right: -20%;
  `};

  ${media.greaterThan('xl')`
    right: -25%;
    width: 120px;
    height: 120px;
  `};

  ${media.greaterThan('xxl')`
    right: -35%;
    width: 135px;
    height: 135px;
  `};
`

export const Slogan = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  background-color: ${PRIMARY_COLOR};
  position: absolute;
  bottom: 0;
  left: 0;
  ${CONTAINER_SPACING.MOBILE}
  width: 85%;
  border-top-right-radius: ${DEFAULT_RADIUS}px;
  min-height: 250px;

  ${media.greaterThan('md')`
    min-height: 350px;
    ${CONTAINER_SPACING.TABLET}
  `};

  ${media.greaterThan('lg')`
    min-height: 400px;
    width: 80%;
    ${CONTAINER_SPACING.SMALL_DESKTOP};
  `};

  ${media.greaterThan('xl')`
    min-height: 400px;
    width: 70%;
    ${CONTAINER_SPACING.DESKTOP};
  `};

  ${media.greaterThan('xxl')`
    min-height: 400px;
    width: 60%;
    ${CONTAINER_SPACING.WIDE};
    padding-right: ${triplePx()};
    padding-top: ${triplePx()};
    padding-bottom: ${triplePx()};
  `};
`

export const Headline = styled.h1`
  color: ${WHITE_COLOR};
  font-size: 1rem;
  padding: ${twoThirdPx()} 0;

  ${media.greaterThan('md')`
    font-size: 1.5rem;
  `};

  ${media.greaterThan('xl')`
    font-size: 1.7rem;
  `};

  ${media.greaterThan('xxl')`
    font-size: 1.9rem;
  `};
`

export const SubHeadline = styled.h2`
  color: ${WHITE_COLOR};
  font-size: 2rem;
  line-height: 1.29;
  width: 95%;

  ${media.greaterThan('md')`
    font-size: 2.5rem;
    width: 92%;
  `};

  ${media.greaterThan('lg')`
    font-size: 3rem;
    width: 90%;
  `};

  ${media.greaterThan('xl')`
    font-size: 3.5rem;
    width: 90%;
  `};
`
