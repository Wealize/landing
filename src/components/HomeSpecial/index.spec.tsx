import React from 'react'
import { mount } from 'enzyme'

import {
  Container,
  Background,
  Button,
  Headline,
  Slogan,
  SubHeadline
} from './styles'
import HomeSpecial from './index'

describe('<HomeSpecial />', () => {
  const wrapper = mount(<HomeSpecial />)

  it('wrapper: should render', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render content correctly', () => {
    expect(wrapper.find(Container)).toHaveLength(1)
    expect(wrapper.find(Background)).toHaveLength(1)
    expect(wrapper.find(Button)).toHaveLength(1)
    expect(wrapper.find(Slogan)).toHaveLength(1)
    expect(wrapper.find(Headline)).toHaveLength(1)
    expect(wrapper.find(SubHeadline)).toHaveLength(1)
    expect(wrapper.find(Headline).text().length).toBeGreaterThan(0)
    expect(wrapper.find(SubHeadline).text().length).toBeGreaterThan(0)
  })
})
