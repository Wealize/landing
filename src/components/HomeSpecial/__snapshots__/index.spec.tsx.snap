// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeSpecial /> wrapper: should render 1`] = `
<Memo(HomeSpecial)>
  <styles__Container>
    <div
      className="styles__Container-gjaj1n-0 fKDDDD"
    >
      <styles__Background>
        <div
          className="styles__Background-gjaj1n-1 cHJHkT"
        />
      </styles__Background>
      <styles__Slogan>
        <div
          className="styles__Slogan-gjaj1n-3 QNtvo"
        >
          <styles__Headline>
            <h1
              className="styles__Headline-gjaj1n-4 lfBZzQ"
            >
              home:slogan.headline
            </h1>
          </styles__Headline>
          <styles__SubHeadline>
            <h2
              className="styles__SubHeadline-gjaj1n-5 kuqDxN"
            >
              home:slogan.subheadline
            </h2>
          </styles__SubHeadline>
          <styles__Button
            aria-label="scroll-button"
            onClick={[Function]}
          >
            <button
              aria-label="scroll-button"
              className="styles__Button-gjaj1n-2 fcDxWJ"
              onClick={[Function]}
            >
              <Memo(RoundArrowIcon)>
                <svg
                  height="6em"
                  viewBox="0 0 90 90"
                  width="6em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M44.999 89.999a45 45 0 1145-45 45 45 0 01-45 45z"
                    fill="#66f"
                  />
                  <path
                    d="M32.513 36.173l11.2-9.05v38.525h2.119V27.162l11.158 9.012 1.336-1.654-12.493-10.09-.415-.335h-1.336l-.368.3-12.54 10.129z"
                    fill="#fff"
                  />
                </svg>
              </Memo(RoundArrowIcon)>
            </button>
          </styles__Button>
        </div>
      </styles__Slogan>
    </div>
  </styles__Container>
</Memo(HomeSpecial)>
`;
